src/planner/operator/logical_mark_accumulate.cpp

```diff
@@ -27,13 +27,10 @@ void LogicalMarkAccumulate::computeFlatSchema() {
 }
 
 f_group_pos_set LogicalMarkAccumulate::getGroupsPosToFlatten() const {
-    f_group_pos_set dependentGroupsPos;
     auto childSchema = children[0]->getSchema();
-    for (auto& key : keys) {
-        for (auto groupPos : childSchema->getDependentGroupsPos(key)) {
-            dependentGroupsPos.insert(groupPos);
-        }
-    }
+    f_group_pos_set dependentGroupsPos = childSchema->getGroupsPosInScope();
+    // TODO(Xiyang/Ziyi): we don't need to flatten all. But hash aggregate seems to not allow
+    // flat key with unFlat payload.
```

Comment:
Yes, i agree with this. The factorized table that guodong meets may be related to this. We are scanning multiple rows from hashaggregate table to a flat vector